flow/project.py

```diff
@@ -291,13 +304,19 @@ def __init__(self, id, name, job, cmd, directives=None):
         self.directives._keys_set_by_user = keys_set_by_user
 
     def __str__(self):
-        return "{}({})".format(self.name, self.job)
+        if len(self._jobs) == 1:
+            return "{}({})".format(self.name, str(self._jobs[0]))
+        else:
+            minimum = self._jobs[0]._project.min_len_unique_id()
+            return "{}-{}: ({}-{})".format(
+                self.name, len(self._jobs), str(self._jobs[0])[:minimum],
+                str(self._jobs[-1])[:minimum])
```

Comment:
I'd prefer to use f-strings whenever possible (readability + performance).

Reply:
Thanks for this suggestion. I'll use f-strings whenever possible.